src/pubd/repository.rs

```diff
@@ -183,60 +521,78 @@ impl RrdpServer {
 
         let notification = Notification::create(session, snapshot_ref);
 
-        Ok(RrdpSessionReset::new(snapshot, notification))
-    }
-
-    fn apply_reset(&mut self, reset: RrdpSessionReset) {
-        let (snapshot, notification) = reset.unpack();
-
         self.serial = notification.serial();
         self.session = notification.session();
         self.notification = notification;
         self.old_notifications.clear();
         self.snapshot = snapshot;
-        self.deltas = vec![];
+        self.deltas = VecDeque::new();
+
+        Ok(())
     }
 
     /// Updates the RRDP server with the elements. Note that this assumes that
     /// the delta has already been checked against the jail and current
-    /// objects of the publisher. Also note that this only becomes effective
-    /// after the corresponding events have been applied.
-    fn publish(&self, elements: DeltaElements) -> Result<RrdpUpdate, Error> {
-        let next = self.serial + 1;
+    /// objects of the publisher.
+    fn publish(
```

Comment:
Does this need, or already have somewhere, locking? I'm wondering if it can end up being invoked concurrently?